### PR TITLE
Onboarding Template Overhaul

### PR DESCRIPTION
**Pull request by JacobJaffe**
After the QA report, there were lots of issues around on-boarding flow screens. Because the files are copied from each other, I transitioned the onboarding pages to use an `OnboardingTemplate`, and turned the permission screens into two separate screens. That should help us in debugging the permission logic. 

image
image
image
image
image
Scrolling now:
image
image
image

iphoneX:
image
image

### CODE REVIEW

File: app/views/onboarding/OnboardingTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  body,
+  primaryButtonLabel,
+  primaryButtonOnPress,
+  secondaryButtonLabel,
+  secondaryButtonOnPress,
+  invertIcon,
+  theme,
+}: OnboardingTemplateProps): JSX.Element => {
+  return (
+    <Theme use={theme === 'light' ? 'default' : 'violet'}>
```
Comment (JacobJaffe) [inline app/views/onboarding/OnboardingTemplate.tsx:43]: @johnschoeman I'm sorry for digging us deeper in this theming hole, this is only being used for light / dark themed buttons. I'd like to refactor the button component soon and remove these themes here & in the export template

File: app/views/onboarding/OnboardingLocations.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React, { useContext } from 'react';
```
Comment (JacobJaffe) [inline app/views/onboarding/OnboardingLocations.tsx:1]: @yonatanmk Check this file out for potential location services issues

**Review by yonatanmk — APPROVED** [review]
LGTM

File: app/views/onboarding/OnboardingNotifications.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  navigation: NavigationScreenProp<NavigationState, NavigationParams>;
+};
+
+const OnboardingNotifications = ({
+  navigation,
+}: OnboardingNotificationsProps): JSX.Element => {
+  const { authSubscription, location } = useContext(PermissionsContext);
+
+  const requestNotifications = async () => {
+    await location.request();
```
Comment (yonatanmk) [inline app/views/onboarding/OnboardingNotifications.tsx]: This should definitely be `notification` not `location`

**Review by yonatanmk — CHANGES_REQUESTED** [review]
Need to update `OnboardingNotifications` per my comment

File: app/views/onboarding/OnboardingNotifications.tsx (hunk 3 above)
Comment (JacobJaffe) [reply app/views/onboarding/OnboardingNotifications.tsx]: good catch yeah

File: app/views/onboarding/OnboardingLocations.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    await location.request();
+    await authSubscription.request();
+    completeOnboarding();
+  };
+
+  const dispatch = useDispatch();
+  const completeOnboarding = () => {
+    // TODO: investigate if this is broken. This is directly copied from our old OnboardingPermissions, but the types look wrong.
+    const storeData = ((location.status ===
+      PermissionStatus.GRANTED) as unknown) as string;
```
Comment (yonatanmk) [inline app/views/onboarding/OnboardingLocations.tsx]:
```suggestion
    const storeData = JSON.stringify(location.status === PermissionStatus.GRANTED);
```
I think the odd types were just added during the conversion to TypeScript. The `SetStoreData` method specifies that its parameters must be strings though it also calls `JSON.stringify` on any arguments that aren't strings

**Review by yonatanmk — APPROVED** [review]
Permissions are working for me once again

File: app/views/onboarding/OnboardingTemplate.tsx (hunk 1 above)
Comment (johnschoeman) [reply app/views/onboarding/OnboardingTemplate.tsx:43]: makes sense to me!

**Review by johnschoeman — APPROVED** [review]
Overall looks great and is quite the improvement!

As a side note, I am less of a fan of the ComponentTemplate pattern as it strikes me as being too dry in the view layer, where you typically want to have more flexibility, but nbd tbh.

**Pull request merged**